### PR TITLE
RHCLOUD-36015: Log/Raise warnings when empty messages are produced

### PR DESCRIPTION
**Pull request by alechenninger**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-36015

## Description of Intent of Change(s)
Saving an empty event to the outbox is inefficient, so this avoids doing that.

However, it is also surprising. In order to identify when these probably unintentional cases are happening, this also logs when it does.

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by coderbydesign — APPROVED** [review]

File: rbac/management/relation_replicator/outbox_replicator.py [hunk 1]
```diff
@@ -39,33 +50,52 @@ def __init__(self, log: Optional["OutboxLog"] = None):
...
+        payload: ReplicationEventPayload,
+        event_type: ReplicationEventType,
+        event_info: dict[str, object],
+        aggregateid: str,
+    ):
         """Save replication event."""
         # TODO: Can we add these as proper fields for kibana but also get logged in simple formatter?
+        logged_info = " ".join([f"info.{key}='{str(value)}'" for key, value in event_info.items()])
+
+        if not payload["relations_to_add"] and not payload["relations_to_remove"]:
```
Comment (lpichler) [inline rbac/management/relation_replicator/outbox_replicator.py:80]: Does this work only when `payload["relations_to_add"] is None` or should it consider also empty array(s) ?

Comment (alechenninger) [reply rbac/management/relation_replicator/outbox_replicator.py:80]: Good question! It should also work with either I think, and I think `None` would be a type error. The tests use empty arrays.

Comment (lpichler) [reply rbac/management/relation_replicator/outbox_replicator.py:80]: That's really cool!

**Pull request merged**